File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/SyncLoanRepaymentAdapter.java
package com.mifos.mifosxdroid.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.mifos.mifosxdroid.databinding.ItemSyncLoanRepaymentBinding;
import com.mifos.objects.PaymentTypeOption;
import com.mifos.objects.accounts.loan.LoanRepaymentRequest;
import com.mifos.utils.Utils;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;


/**
 * Created by Rajan Maurya on 30/07/16.
 */
public class SyncLoanRepaymentAdapter extends
        RecyclerView.Adapter<SyncLoanRepaymentAdapter.ViewHolder> {

    private List<LoanRepaymentRequest> loanRepaymentRequests;
    private List<PaymentTypeOption> mPaymentTypeOptions;

    @Inject
    public SyncLoanRepaymentAdapter() {
        loanRepaymentRequests = new ArrayList<>();
        mPaymentTypeOptions = new ArrayList<>();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        ItemSyncLoanRepaymentBinding binding = ItemSyncLoanRepaymentBinding
                .inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

        LoanRepaymentRequest loanRepaymentRequest = loanRepaymentRequests.get(position);
        String paymentTypeName =
                Utils.getPaymentTypeName(Integer.parseInt(
                        loanRepaymentRequest.getPaymentTypeId()), mPaymentTypeOptions);

        holder.tv_loan_id.setText(String.valueOf(loanRepaymentRequest.getLoanId()));
        holder.tv_account_number.setText(loanRepaymentRequest.getAccountNumber());

        if (mPaymentTypeOptions.size() != 0) {
            holder.tv_payment_type.setText(paymentTypeName);
        }

        holder.tv_transaction_amount.setText(loanRepaymentRequest.getTransactionAmount());
        holder.tv_transaction_date.setText(loanRepaymentRequest.getTransactionDate());

        if (loanRepaymentRequest.getErrorMessage() != null) {
            holder.tv_error_message.setText(loanRepaymentRequest.getErrorMessage());
            holder.tv_error_message.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public int getItemCount() {
        return loanRepaymentRequests.size();
    }

    public void setLoanRepaymentRequests(List<LoanRepaymentRequest> repaymentRequests) {
        loanRepaymentRequests = repaymentRequests;
        notifyDataSetChanged();
    }

    public void setPaymentTypeOptions(List<PaymentTypeOption> paymentTypeOptions) {
        mPaymentTypeOptions = paymentTypeOptions;
        notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        TextView tv_loan_id;

        TextView tv_account_number;

        TextView tv_payment_type;

        TextView tv_transaction_amount;

        TextView tv_transaction_date;

        TextView tv_error_message;

        public ViewHolder(ItemSyncLoanRepaymentBinding binding) {
            super(binding.getRoot());

            tv_loan_id = binding.tvDbLoanId;
            tv_account_number = binding.tvDbAccountNumber;
            tv_payment_type = binding.tvDbPaymentType;
            tv_transaction_amount = binding.tvDbTransactionAmount;
            tv_transaction_date = binding.tvDbTransactionDate;
            tv_error_message = binding.tvErrorMessage;
        }
    }
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/SyncLoanRepaymentAdapter.kt
package com.mifos.mifosxdroid.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.mifos.core.objects.accounts.loan.LoanRepaymentRequest
import com.mifos.mifosxdroid.databinding.ItemSyncLoanRepaymentBinding
import com.mifos.utils.Utils.getPaymentTypeName
import javax.inject.Inject

/**
 * Created by Rajan Maurya on 30/07/16.
 */
class SyncLoanRepaymentAdapter @Inject constructor() :
    RecyclerView.Adapter<SyncLoanRepaymentAdapter.ViewHolder>() {
    private var loanRepaymentRequests: List<LoanRepaymentRequest>
    private var mPaymentTypeOptions: List<com.mifos.core.objects.PaymentTypeOption>

    init {
        loanRepaymentRequests = ArrayList()
        mPaymentTypeOptions = ArrayList()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ItemSyncLoanRepaymentBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val loanRepaymentRequest = loanRepaymentRequests[position]
        val paymentTypeName = loanRepaymentRequest.paymentTypeId?.let {
            getPaymentTypeName(
                it.toInt(), mPaymentTypeOptions
            )
        }
        holder.binding.tvLoanId.text = loanRepaymentRequest.loanId.toString()
        holder.binding.tvAccountNumber.text = loanRepaymentRequest.accountNumber
        if (mPaymentTypeOptions.isNotEmpty()) {
            holder.binding.tvPaymentType.text = paymentTypeName
        }
        holder.binding.tvTransactionAmount.text = loanRepaymentRequest.transactionAmount
        holder.binding.tvTransactionDate.text = loanRepaymentRequest.transactionDate
        if (loanRepaymentRequest.errorMessage != null) {
            holder.binding.tvErrorMessage.text = loanRepaymentRequest.errorMessage
            holder.binding.tvErrorMessage.visibility = View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return loanRepaymentRequests.size
    }

    fun setLoanRepaymentRequests(repaymentRequests: List<LoanRepaymentRequest>) {
        loanRepaymentRequests = repaymentRequests
        notifyDataSetChanged()
    }

    fun setPaymentTypeOptions(paymentTypeOptions: List<com.mifos.core.objects.PaymentTypeOption>) {
        mPaymentTypeOptions = paymentTypeOptions
        notifyDataSetChanged()
    }

    class ViewHolder(val binding: ItemSyncLoanRepaymentBinding) :
        RecyclerView.ViewHolder(binding.root)
}